### ISSUE TITLE
self signed certificate in certificate chai

### ISSUE THREAD

**Issue by TomInIowa** [philippelt/netatmo-api-python#54]
I loved your program ... been using for years.   Now it quit a few days ago.
Trying to get the new version  with client_id, client_secret and refresh_token to work.

In lnetatmo.py   I changed just 3 things ....   client_id ,   client_secret,   refresh_token

python3  lnetatmo.py ( on mac computer using terminal )

urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)

[error3.txt](https://github.com/philippelt/netatmo-api-python/files/12063141/error3.txt)

**Comment by AtsuMiyai** [comment]
I think that you need to update your lnetatomo version.

**Comment by philippelt** [comment]
This error is weird, looks like you are not talking to the right server.

Could you check the content of ```_BASE_URL``` ? it should be https://api.netatmo.com/
if you enter in a terminal the command ```dig +short api.netatmo.com``` you should get the following results
```
front-azure.netatmo.net.
51.145.143.28
```
And if you try the command ```curl -i https://api.netatmo.com/oauth2/token``` you should get the following results
```
HTTP/1.1 400
Server: nginx
Date: Sun, 16 Jul 2023 09:10:43 GMT
Content-Type: application/json
Transfer-Encoding: chunked
Connection: keep-alive
Expires: 0
Cache-Control: no-cache, must-revalidate
X-XSS-Protection: 1; mode=block
Access-Control-Allow-Origin: *
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Powered-By: Netatmo

{"error":"invalid_request","error_description":"Invalid grant_type parameter or parameter missing"}
```
The error here is normal, the goal of the test is just to check if you are reaching the appropriate server and that the TLS handshake was successfull.

Check that you don't have a proxy enabled in your network config. Go to network config (clicking on the wifi icon on top right), tcp/ip parameters and Proxy setting. If you are a basic home user, you should not have any proxy enabled...

**Comment by TomInIowa** [comment]
Thanks for the quick response.  I tried the things you suggested... got the response you said I should.

I moved to a different mac computer .. went to
https://github.com/philippelt/netatmo-api-python
[https://opengraph.githubassets.com/923063e30c0724422a47b0fdedae0e71a3b052fa7be01630f8097b479a435a28/philippelt/netatmo-api-python]<https://github.com/philippelt/netatmo-api-python>
GitHub - philippelt/netatmo-api-python: Netatmo connect API python client (for Netatmo information, see https://dev.netatmo.com)<https://github.com/philippelt/netatmo-api-python>
Netatmo connect API python client (for Netatmo information, see https://dev.netatmo.com) - GitHub - philippelt/netatmo-api-python: Netatmo connect API python client (for Netatmo information, see ht...
github.com

and downloaded the files.  Change just three lines.
NOW IT WORKS!!!

Thank you so much.  Great coding.
________________________________

**Comment by philippelt** [comment]
Great to know you solved the problem.

Don't forget, if you finally locate the original cause to add a comment to explain what happened as it may help someone else facing same kind of issue !

**Comment by TomInIowa** [comment]
I copied the lnetatmo.py file  (that  works on one mac computer) back to the mac I started on.
It does NOT work.   I have something messed up on it.

**Comment by philippelt** [comment]
Did you check the proxy setting ? I have seen in the past a Mac infected by a malware that implemented a proxy to capture ip traffic...

This could lead to such error

**Comment by TomInIowa** [comment]
I was running python 3.8

I  removed it

Had python 3.7 left

tried it..... IT works

install python 3.11   and it quit working

**Comment by philippelt** [comment]
You said *it quit working*, do you mean it worked ?

Good news it was just a mess in the python install, easy to sort out...

**Comment by philippelt** [comment]
I am using it with python 3.10. It should work fine with python 3.11

enjoy

**Comment by TomInIowa** [comment]
I installed python 3.11 two different ways .... once using homebrew (which took a long time) and. once from python org website.    Both times ,     lnetatmo.py   quit working.
I'm on mac computer running  Ventura 13.4.1
Going back to python 3.7. (if I can)

**Comment by philippelt** [comment]
Interesting, if I can find some time, I will give it a try on my Mac tomorrow. Usually, I install from python.org

**Comment by philippelt** [comment]
I just did the test... and got the same result as you at the first run.

This is due to a restriction in the python installer. It doesn't update the root certificates for python openssl.
There is a shell script provided by the installer : /Applications/Python\ 3.11/Install\ Certificates.command
Just run it and it will fix the problem !

I should have remember that, I did that also for the previous python mac install 😄

But today, for most computer/programming work, I am using an Ubuntu Desktop workstation and server thus I am less frequently doing things like that on my Macs that are mainly used for office/photo work.

P.S. For more details, look at https://github.com/actions/setup-python/issues/512 and https://github.com/actions/python-versions/pull/189
